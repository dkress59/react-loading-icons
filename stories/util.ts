import { ArgTypes } from '@storybook/react'
import { SVGProps } from 'react'

export type IconProps = JSX.IntrinsicAttributes & SVGProps<SVGSVGElement>
export const IconFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: 'transparent',
	strokeOpacity: 1,
	width: 'auto',
}
export const IconStroke: IconProps = {
	fill: 'transparent',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}
export const IconStrokeFill: IconProps = {
	fill: '#06bcee',
	fillOpacity: 1,
	height: '3em',
	speed: 1,
	stroke: '#06bcee',
	strokeOpacity: 1,
	strokeWidth: 2,
	width: 'auto',
}

export const IconFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: 'transparent',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: 'transparent',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
export const IconStrokeFillArgs: ArgTypes = {
	height: {
		name: 'Height',
		control: 'text',
		defaultValue: '3em',
	},
	width: {
		name: 'Width',
		control: 'text',
		defaultValue: 'auto',
	},
	fill: {
		name: 'Fill',
		control: 'color',
		defaultValue: '#06bcee',
	},
	stroke: {
		name: 'Stroke',
		control: 'color',
		defaultValue: '#06bcee',
	},
	speed: {
		name: 'Speed',
		control: 'number',
		defaultValue: 1,
	},
}
