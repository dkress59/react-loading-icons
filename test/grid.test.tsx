import { render, screen } from '@testing-library/react'
import React from 'react'

import Grid from '../src/components/grid'
import { testProps } from './util'

describe('<Grid /> component', () => {
	it('renders and matches snapshot', () => {
		render(<Grid data-testid="icon" />)
		const svg = screen.getByTestId('icon')
		expect(svg).not.toBeNull()
		expect(svg).toMatchSnapshot()
		expect(svg.getAttributeNames()).toContain('viewBox')
		expect(svg.getAttribute('class')).toBe('icon-loading')
	})
	it('inherits props', () => {
		render(<Grid {...testProps} />)
		const svg = screen.getByTestId('icon')
		expect(svg.getAttribute('height')).toBe('2em')
		expect(svg.getAttribute('width')).toBe('auto')
		expect(svg.getAttribute('class')).toBe('icon-loading test')
	})
})
