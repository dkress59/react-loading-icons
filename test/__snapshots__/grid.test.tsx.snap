// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Grid /> component renders and matches snapshot 1`] = `
<svg
  class="icon-loading"
  data-testid="icon"
  height="105"
  viewBox="0 0 105 105"
  width="105"
>
  <circle
    cx="12.5"
    cy="12.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="0s"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="12.5"
    cy="52.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="100ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="52.5"
    cy="12.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="300ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="52.5"
    cy="52.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="600ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="92.5"
    cy="12.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="800ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="92.5"
    cy="52.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="400ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="12.5"
    cy="92.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="700ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="52.5"
    cy="92.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="500ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
  <circle
    cx="92.5"
    cy="92.5"
    fill="#fff"
    r="12.5"
  >
    <animate
      attributeName="fill-opacity"
      begin="200ms"
      calcMode="linear"
      dur="1s"
      repeatCount="indefinite"
      values="1;.2;1"
    />
  </circle>
</svg>
`;
